src/sknnr_spatial/estimator.py

```diff
@@ -187,6 +217,8 @@ def kneighbors(
         neigh_ind : Numpy or Xarray image with 3 dimensions (y, x, neighbor)
             Indices of the nearest points in the population matrix.
         """
+        kneighbors_kwargs["return_distance"] = return_distance
+
```

Comment:
Oh, interesting.  Was this a silent bug?  That is, `return_distance` wasn't being passed to the estimator before now?  That totally went past me before.  Sorry!

Reply:
No, `return_distance` used to just be implicitly passed through `kneighbor_kwargs` (can't remember exactly why I made that choice), but in order to define the `@overload` signatures based on `return_distance` I had to make it explicit. I left it implicit in the image wrapper `kneighbors` methods here, but maybe it should just be explicit everywhere to match the `sklearn` signature?

Reply:
Sorry, I had read the previous code incorrectly - now I see what you're saying.  

To match the `sklearn` signature, would you also make the `n_neighbors` keyword explicit as well?  I guess that's  the only other keyword argument.

And, just to be annoying :wink:, we probably want to add `return_distance` as a separate entry in the docstring rather than lumping in `**kneighbors_kwargs`.  It's a lot of work making things better!

Reply:
> To match the sklearn signature, would you also make the n_neighbors keyword explicit as well?

Yes! I also just realized that `DaskBackedWrapper.kneighbors` needs to set the output shape based on `n_neighbors` if it's passed in! Currently it always assumes the output shape is the number of neighbors used to instantiate the estimator, so it would break with any other value. I'll probably just make that fix here.

> And, just to be annoying 😉, we probably want to add return_distance as a separate entry in the docstring rather than lumping in **kneighbors_kwargs.

Totally, thanks for catching my oversights!